diesel/src/query_builder/ast_pass.rs

```diff
@@ -46,15 +56,41 @@ impl<'a, DB> AstPass<'a, DB> where
         }
     }
 
+    pub fn push_sql(&mut self, sql: &str) {
+        if let AstPassInternals::ToSql(ref mut builder) = self.internals {
+            builder.push_sql(sql);
+        }
+    }
```

Comment:
I like these methods that are no-ops for most variants, by the way. Assuming rustc can track which variant we are looking at, this should also be an easy target for dead code elimination.